test/e2e/settings/customize_health_check_path.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package settings
+
+import (
+	"net/http"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/parnurzeal/gorequest"
+
+	"k8s.io/ingress-nginx/test/e2e/framework"
+)
+
+var _ = framework.IngressNginxDescribe("Customize health check path", func() {
+	f := framework.NewDefaultFramework("custom-health-check-path")
+
+	Context("with a plain HTTP ingress", func() {
+		It("should return HTTP/1.1 200 OK on custom health check path and port", func() {
+
+			f.WaitForNginxConfiguration(func(server string) bool {
+				return strings.Contains(server, "location /not-healthz")
+			})
+
+			resp, body, errs := gorequest.New().
+				Get(f.GetURL(framework.HTTP) + ":9090/not-healthz").
```

Comment:
Use `fmt.Sprintf("http://%v:9090/not-healthz", f.GetNginxIP())` (f.GetURL retursn protocol + IP + port)

Reply:
`f.GetURL()` uses `f.GetNginxIP()` anyway
and according to logs of the e2e test failing the url seems to be fine: `http://10.108.242.250:9090/not-healthz`

the problem is that while the health check works properly from the k8s master the port is not exposed to host where the e2e binary is run, so I cannot probe it with a http request

checking for `framework.WaitForPodsReady` should be the right solution. that is done anyway on the `BeforeEach` so it could be redudant to be added a second time in scenario.
what do you think?

Reply:
> what do you think?

Yes, this is a lot simpler.